engine-transactions/src/lib.rs

```diff
@@ -117,12 +120,12 @@ impl From<EthTransactionKind> for NormalizedEthTransaction {
                 data: tx.transaction.data,
                 access_list: tx.transaction.access_list,
             },
-        }
+        })
     }
 }
 
 impl NormalizedEthTransaction {
-    pub fn intrinsic_gas(&self, config: &evm::Config) -> Option<u64> {
+    pub fn intrinsic_gas(&self, config: &evm::Config) -> Result<u64, ParseTransactionError> {
```

Comment:
Again I think a separate error enum might make sense here. This function is not really about parsing, it's about another validation step.